example/issue_comment.go

```diff
@@ -42,46 +65,187 @@ func (h *PRCommentHandler) Handle(ctx context.Context, eventType, deliveryID str
+		return fmt.Errorf("failed to get JUnitTestSuites from the file %s: %+v", junitFilename, err)
+	}
+
+	failedTCReport := setHeaderString(logger, overallJUnitSuites)
+	failedTCReport.extractFailedTestCases(logger, overallJUnitSuites)
+
+	failedTCReport.updateCommentWithFailedTestCasesReport(ctx, logger, client, event, body)
+
+	return nil
+}
+
+// extractProwJobURLFromCommentBody extracts the
+// Prow job's URL from the given PR comment's body
+func extractProwJobURLFromCommentBody(logger zerolog.Logger, commentBody string) (string, error) {
+	r, _ := regexp.Compile(regexToFetchProwURL)
+	sliceOfMatchingString := r.FindStringSubmatch(commentBody)
+	if sliceOfMatchingString == nil {
+		return "", fmt.Errorf("regex string %s found no matches for the comment body: %s", regexToFetchProwURL, commentBody)
+	}
+	prowJobURL := sliceOfMatchingString[1]
+	logger.Debug().Msgf("Prow Job's URL: %s", prowJobURL)
+
+	return prowJobURL, nil
+}
+
+// getTestSuitesFromXMLFile returns all the JUnitTestSuites
+// present within a file with the given name
+func getTestSuitesFromXMLFile(scanner *prow.ArtifactScanner, logger zerolog.Logger, filename string) (*reporters.JUnitTestSuites, error) {
+	overallJUnitSuites := &reporters.JUnitTestSuites{}
+
+	for _, artifactsFilenameMap := range scanner.ArtifactStepMap {
+		for artifactFilename, artifact := range artifactsFilenameMap {
+			if string(artifactFilename) == filename {
+				if err := xml.Unmarshal([]byte(artifact.Content), overallJUnitSuites); err != nil {
+					logger.Error().Err(err).Msg("cannot decode JUnit suite into xml")
+					return &reporters.JUnitTestSuites{}, err
+				}
+				return overallJUnitSuites, nil
+			}
+		}
+	}
+
+	return &reporters.JUnitTestSuites{}, fmt.Errorf("couldn't find the %s file", filename)
+}
+
+// setHeaderString initialises struct FailedTestCasesReport's
+// 'headerString' field based on phase at which Prow job failed
+func setHeaderString(logger zerolog.Logger, overallJUnitSuites *reporters.JUnitTestSuites) *FailedTestCasesReport {
+	failedTCReport := FailedTestCasesReport{}
+
+	if len(overallJUnitSuites.TestSuites) == 0 {
+		logger.Debug().Msg("The given Prow job failed while creating the cluster")
+		failedTCReport.headerString = ":rotating_light: **Error occurred while creating the cluster, please check the Prow's build logs.**\n"
+	} else if len(overallJUnitSuites.TestSuites) == 1 && overallJUnitSuites.TestSuites[0].Name == openshiftCITestSuiteName {
+		logger.Debug().Msg("The given Prow job failed during bootstrapping the cluster")
+		failedTCReport.hasBootstrapFailure = true
+		failedTCReport.headerString = ":rotating_light: **Error occurred during the cluster's Bootstrapping phase, list of failed Spec(s)**: \n"
+	} else {
+		logger.Debug().Msg("The given Prow job failed while running the E2E tests")
+		failedTCReport.headerString = ":rotating_light: **Error occurred while running the E2E tests, list of failed Spec(s)**: \n"
+	}
```

Comment:
Here's how the Prow job for these 3 `if`, `else if`, and `else` would be processed by the GitHub and displayed to the users:
* [Example](https://github.com/dheerajodha/sample-github-app/pull/5#issuecomment-1898171606), when the Prow job failed during cluster creation.
* [Example](https://github.com/dheerajodha/sample-github-app/pull/5#issuecomment-1898175959),  when the Prow job failed while bootstrapping (or installing Konflux on) the cluster -- lmk if I should change the phrasing.
* When the Prow job failed while running the E2E tests:
** [where the E2E test](https://github.com/dheerajodha/sample-github-app/pull/5#issuecomment-1898180007) **failed**
** [where the E2E test](https://github.com/dheerajodha/sample-github-app/pull/5#issuecomment-1898177916) **panicked**

Reply:
@psturc @dheerajodha For the example when the prow job failed while boostrapping the cluster, what are your thoughts on looking at the system-error element of that test case and showing like the last 12-15 lines? Rather than looking at the failure-message element? I didn't really find that message of `redhat-appstudio-e2e has failed` helpful for that use case.

Reply:
yep, I think that is a good idea

Reply:
Are we going to include the changes I mentioned here as part of this PR or no?

Reply:
I'll include them as part of this PR, thanks for the suggestion!

Reply:
Addressed [here](https://github.com/dheerajodha/go-githubapp/pull/1/commits/e9250f3633acda69976b66d49f5df6c81d929a16)